### ISSUE TITLE
Problem about the evaluation.

### ISSUE THREAD

**Issue by pengshuang** [MiuLab/FlowDelta#6]
Hi,

Thanks for your code. 

I have used your code to generate a submission file, but the officer from the QuAC told me that I accidentally uses future questions and answers in a dialogue. Because they see the different results using the normal test set and the randomized one.

Did you run into the same problem ?

Shuang

**Comment by pengshuang** [comment]
FYI, I use the model with option **--no flow**

**Comment by exe1023** [comment]
Hi, 
I didn't run into this problem when submitting to the public leaderboard. 
Maybe you can also use the randomized dev set to find the problem?

**Comment by pengshuang** [comment]
Hi.
Thanks for the reply.
I also use the randomized dev set and find the problem.

Here if I remove the context features, the problem is solved. 
https://github.com/MiuLab/FlowDelta/blob/845aa8cc2a6804a3f663cfbe8ffdf09710323c44/BertFlowDelta/run_quac.py#L825

This is so strange and I check it for a long time (but no results).
Do you have any suggestions ?

**Comment by pengshuang** [comment]
Hi @exe1023 

I test your origin code with the randomized dev set **test1.json** and **test2.json**. Sadly, I also get different results.

FYI
test1.json
`{"data": [{"paragraphs": [{"context": "In May 1983, she married Nikos Karvelas, a composer, with whom she collaborated in 1975 and in November she gave birth to her daughter Sofia. After their marriage, she started a close collaboration with Karvelas. Since 1975, all her releases have become gold or platinum and have included songs by Karvelas. In 1986, she participated at the Cypriot National Final for Eurovision Song Contest with the song Thelo Na Gino Star (\"I Want To Be A Star\"), taking second place. This song is still unreleased up to date. In 1984, Vissi left her record company EMI Greece and signed with CBS Records Greece, which later became Sony Music Greece, a collaboration that lasted until 2013. In March 1984, she released Na 'Hes Kardia (\"If You Had a Heart\"). The album was certified gold. The following year her seventh album Kati Simveni (\"Something Is Happening\") was released which included one of her most famous songs, titled \"Dodeka\" [\"Twelve (O'Clock)\"] and reached gold status selling 80.000 units. In 1986 I Epomeni Kinisi (\"The Next Move\") was released. The album included the hit Pragmata (\"Things\") and went platinum, becoming the best selling record of the year. In February 1988 she released her ninth album Tora (\"Now\") and in December the album Empnefsi! (\"Inspiration!\") which went gold. In 1988, she made her debut as a radio producer on ANT1 Radio. Her radio program was titled after one of her songs Ta Koritsia Einai Atakta (\"Girls Are Naughty\") and was aired every weekend. In the same year, she participated with the song Klaio (\"I'm Crying\") at the Greek National Final for Eurovision Song Contest, finishing third. In 1989, she released the highly successful studio album Fotia (Fire), being one of the first albums to feature western sounds. The lead single Pseftika (\"Fake\") became a big hit and the album reached platinum status, selling 180.000 copies and becoming the second best selling record of 1990. She performed at \"Diogenis Palace\" in that same year, Athens's biggest nightclub/music hall at the time. CANNOTANSWER", "qas": [{"followup": "y", "yesno": "x", "question": "what happened in 1983?", "answers": [{"text": "In May 1983, she married Nikos Karvelas,", "answer_start": 0}, {"text":"In May 1983, she married Nikos Karvelas, a composer,", "answer_start": 0}, {"text": "In May 1983, she married Nikos Karvelas,", "answer_start": 0}, {"text": "In May 1983, she married Nikos Karvelas, a composer,", "answer_start": 0}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#0", "orig_answer": {"text": "In May 1983, she married Nikos Karvelas, a composer,", "answer_start": 0}}, {"followup": "m", "yesno": "y", "question": "did they have any children?", "answers": [{"text": "she gave birth to her daughter Sofia.", "answer_start": 104}, {"text": "in 1975 and in November she gave birth to her daughter Sofia.", "answer_start": 80}, {"text": "her daughter Sofia.", "answer_start": 122}, {"text": "in November she gave birth to her daughter Sofia.", "answer_start": 92}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#1", "orig_answer": {"text": "in November she gave birth to her daughter Sofia.", "answer_start": 92}}, {"followup": "n", "yesno": "x", "question": "did she have any other children?", "answers": [{"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#2", "orig_answer": {"text": "CANNOTANSWER", "answer_start": 2024}}, {"followup": "m", "yesno": "x", "question": "what collaborations did she do with nikos?", "answers": [{"text": "After their marriage, she started a close collaboration with Karvelas. Since 1975, all her releases have become gold or platinum and have included songs by Karvelas.", "answer_start": 142}, {"text": "a composer, with whom she collaborated in 1975", "answer_start": 41}, {"text": "Thelo Na Gino Star (\"", "answer_start": 406}, {"text": "Since 1975, all her releases have become gold or platinum and have included songs by Karvelas.", "answer_start": 213}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#3", "orig_answer": {"text": "Since 1975, all her releases have become gold or platinum and have included songs by Karvelas.", "answer_start": 213}}, {"followup": "n", "yesno": "x", "question": "what influences does he have in her music?", "answers": [{"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#4", "orig_answer": {"text": "CANNOTANSWER", "answer_start": 2024}}, {"followup": "y", "yesno": "x", "question": "what were some of the songs?", "answers": [{"text": "Thelo Na Gino Star (\"I Want To Be A Star\"),", "answer_start": 406}, {"text": "Thelo Na Gino Star (\"", "answer_start": 406}, {"text": "one of her most famous songs, titled \"Dodeka\" [\"Twelve (O'Clock)\"]", "answer_start": 879}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#5", "orig_answer": {"text": "one of her most famous songs, titled \"Dodeka\" [\"Twelve (O'Clock)\"]", "answer_start": 879}}, {"followup": "y", "yesno": "x", "question": "how famous was it?", "answers": [{"text": "reached gold status selling 80.000 units.", "answer_start": 950}, {"text": "reached gold status selling 80.000 units.", "answer_start": 950}, {"text": "reached gold status selling 80.000 units.", "answer_start": 950}, {"text": "reached gold status selling 80.000 units.", "answer_start": 950}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#6", "orig_answer": {"text": "reached gold status selling 80.000 units.", "answer_start": 950}}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1"}], "section_title": "1983-1989: Collaboration with Nikos Karvelas", "background": "Anna Vissi (Greek: Anna Bisse (pronounced ['ana 'visi], locally ['an:a 'viS:i]); born 20 December 1957), also known as Anna Vishy, is a Greek Cypriot singer, songwriter, actress, television presenter, radio personality, and businesswoman.", "title": "Anna Vissi"}]}`

test2.json
`{"data": [{"paragraphs": [{"context": "In May 1983, she married Nikos Karvelas, a composer, with whom she collaborated in 1975 and in November she gave birth to her daughter Sofia. After their marriage, she started a close collaboration with Karvelas. Since 1975, all her releases have become gold or platinum and have included songs by Karvelas. In 1986, she participated at the Cypriot National Final for Eurovision Song Contest with the song Thelo Na Gino Star (\"I Want To Be A Star\"), taking second place. This song is still unreleased up to date. In 1984, Vissi left her record company EMI Greece and signed with CBS Records Greece, which later became Sony Music Greece, a collaboration that lasted until 2013. In March 1984, she released Na 'Hes Kardia (\"If You Had a Heart\"). The album was certified gold. The following year her seventh album Kati Simveni (\"Something Is Happening\") was released which included one of her most famous songs, titled \"Dodeka\" [\"Twelve (O'Clock)\"] and reached gold status selling 80.000 units. In 1986 I Epomeni Kinisi (\"The Next Move\") was released. The album included the hit Pragmata (\"Things\") and went platinum, becoming the best selling record of the year. In February 1988 she released her ninth album Tora (\"Now\") and in December the album Empnefsi! (\"Inspiration!\") which went gold. In 1988, she made her debut as a radio producer on ANT1 Radio. Her radio program was titled after one of her songs Ta Koritsia Einai Atakta (\"Girls Are Naughty\") and was aired every weekend. In the same year, she participated with the song Klaio (\"I'm Crying\") at the Greek National Final for Eurovision Song Contest, finishing third. In 1989, she released the highly successful studio album Fotia (Fire), being one of the first albums to feature western sounds. The lead single Pseftika (\"Fake\") became a big hit and the album reached platinum status, selling 180.000 copies and becoming the second best selling record of 1990. She performed at \"Diogenis Palace\" in that same year, Athens's biggest nightclub/music hall at the time. CANNOTANSWER", "qas": [{"followup": "y", "yesno": "x", "question": "what happened in 1983?", "answers": [{"text": "In May 1983, she married Nikos Karvelas,", "answer_start": 0}, {"text":"In May 1983, she married Nikos Karvelas, a composer,", "answer_start": 0}, {"text": "In May 1983, she married Nikos Karvelas,", "answer_start": 0}, {"text": "In May 1983, she married Nikos Karvelas, a composer,", "answer_start": 0}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#0", "orig_answer": {"text": "In May 1983, she married Nikos Karvelas, a composer,", "answer_start": 0}}, {"followup": "m", "yesno": "y", "question": "did they have any children?", "answers": [{"text": "she gave birth to her daughter Sofia.", "answer_start": 104}, {"text": "in 1975 and in November she gave birth to her daughter Sofia.", "answer_start": 80}, {"text": "her daughter Sofia.", "answer_start": 122}, {"text": "in November she gave birth to her daughter Sofia.", "answer_start": 92}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#1", "orig_answer": {"text": "in November she gave birth to her daughter Sofia.", "answer_start": 92}}, {"followup": "n", "yesno": "x", "question": "did she have any other children?", "answers": [{"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#2", "orig_answer": {"text": "CANNOTANSWER", "answer_start": 2024}}, {"followup": "m", "yesno": "x", "question": "what collaborations did she do with nikos?", "answers": [{"text": "After their marriage, she started a close collaboration with Karvelas. Since 1975, all her releases have become gold or platinum and have included songs by Karvelas.", "answer_start": 142}, {"text": "a composer, with whom she collaborated in 1975", "answer_start": 41}, {"text": "Thelo Na Gino Star (\"", "answer_start": 406}, {"text": "Since 1975, all her releases have become gold or platinum and have included songs by Karvelas.", "answer_start": 213}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#3", "orig_answer": {"text": "Since 1975, all her releases have become gold or platinum and have included songs by Karvelas.", "answer_start": 213}}, {"followup": "n", "yesno": "x", "question": "what influences does he have in her music?", "answers": [{"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}, {"text": "CANNOTANSWER", "answer_start": 2024}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#4", "orig_answer": {"text": "CANNOTANSWER", "answer_start": 2024}}, {"followup": "y", "yesno": "x", "question": "what were some of the songs?", "answers": [{"text": "Thelo Na Gino Star (\"I Want To Be A Star\"),", "answer_start": 406}, {"text": "Thelo Na Gino Star (\"", "answer_start": 406}, {"text": "one of her most famous songs, titled \"Dodeka\" [\"Twelve (O'Clock)\"]", "answer_start": 879}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#5", "orig_answer": {"text": "one of her most famous songs, titled \"Dodeka\" [\"Twelve (O'Clock)\"]", "answer_start": 879}}, {"followup": "y", "yesno": "x", "question": "how famous was it?", "answers": [{"text": "Thelo Na Gino Star (\"I Want To Be A Star\"),", "answer_start": 406}, {"text": "Thelo Na Gino Star (\"", "answer_start": 406}, {"text": "one of her most famous songs, titled \"Dodeka\" [\"Twelve (O'Clock)\"]", "answer_start": 879}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1_q#6", "orig_answer": {"text": "one of her most famous songs, titled \"Dodeka\" [\"Twelve (O'Clock)\"]", "answer_start": 879}}], "id": "C_5ab583f64dbb47b995cf59328ea0af43_1"}], "section_title": "1983-1989: Collaboration with Nikos Karvelas", "background": "Anna Vissi (Greek: Anna Bisse (pronounced ['ana 'visi], locally ['an:a 'viS:i]); born 20 December 1957), also known as Anna Vishy, is a Greek Cypriot singer, songwriter, actress, television presenter, radio personality, and businesswoman.", "title": "Anna Vissi"}]}`

**Comment by pengshuang** [comment]
The answers to the question **C_5ab583f64dbb47b995cf59328ea0af43_1_q#6** are different.

We just randomize the original answer to this question.

**Comment by menggehe** [comment]
what the F1 score  you got on QuAC ? I still can not get 67.

**Comment by pengshuang** [comment]
> what the F1 score you got on QuAC ? I still can not get 67.

I got 67 but when I try to submit it to the public leaderboard, the officer from the QuAC told me that I accidentally uses future questions and answers in a dialogue.

**Comment by menggehe** [comment]
> > what the F1 score you got on QuAC ? I still can not get 67.
> 
> I got 67 but when I try to submit it to the public leaderboard, the officer from the QuAC told me that I accidentally uses future questions and answers in a dialogue.

Thank you, I'm sorry that i also can't solve this problem.

**Comment by menggehe** [comment]
> > what the F1 score you got on QuAC ? I still can not get 67.
> 
> I got 67 but when I try to submit it to the public leaderboard, the officer from the QuAC told me that I accidentally uses future questions and answers in a dialogue.

Hi，i have som questions.
Is the code you used  the author updated a few days ago？
(such as : max_seq_length=384, batch_size=32, learning_rate=5e-5, epoch=3, ..... )
And the BERT model you used is base or large?

I've been obsessed with this question for a long time, i would appreciate it if you can help me!

**Comment by pengshuang** [comment]
bert-base

We use the code the author updated a few days ago.

**Comment by menggehe** [comment]
> bert-base
> 
> We use the code the author updated a few days ago.

Thank you, i will try it again!

**Comment by pengshuang** [comment]
> > bert-base
> > We use the code the author updated a few days ago.
> 
> Thank you, i will try it again!

But I still think there may be some problem when submitted to the QuAC leaderboard.

**Comment by ady-yu** [comment]
> > > bert-base
> > > We use the code the author updated a few days ago.
> > 
> > 
> > Thank you, i will try it again!
> 
> But I still think there may be some problem when submitted to the QuAC leaderboard.

Hi, do you know why this question occurs. I think the logic in get_context_feature matters.
